src/tokens/TokenDetails.tsx

```diff
@@ -136,7 +136,21 @@ function PriceInfo({ token }: { token: TokenBalance }) {
   )
 }
 
-function Actions({ token }: { token: TokenBalance }) {
+export const onPressCicoAction = (token: TokenBalance, flow: CICOFlow) => {
+  const tokenSymbol = token.symbol
+  // this should always be true given that we only show Add / Withdraw if a
+  // token is CiCoCurrency, but adding it here to ensure type safety
+  if (isCicoToken(tokenSymbol)) {
+    navigate(Screens.FiatExchangeAmount, {
+      currency: tokenSymbol,
+      tokenId: token.tokenId,
+      flow,
+      network: Network.Celo, // TODO: use networkId from token
+    })
+  }
+}
+
+export const useActions = (token: TokenBalance) => {
```

Comment:
Made this a hook that takes a token and returns an array of actions, curious to hear others opinions of if this is needed / good practice.

Reply:
I think this would result in re-computing the list of actions in both the actions component and the bottom sheet. One option might be to compute the list of eligible actions in the Action component behind a use memo and then pass a list of `TokenDetailActionName`s as a prop to the bottom sheet screen. And then maybe we can also have a map of `TokenDetailActionName` to `TokenDetailsAction` which can be used to lookup other properties for rendering